honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,330 @@
+    """
+    return Point3D(point.X, point.Y, point.Z)
+
+
+def to_vector3d(vector):
+    """Create a Ladybug Vector3D from a rhino3dm Vector3d.
+
+    Args:
+        vector: A rhino3dm vector3d.
+
+    Returns:
+         A Ladybug Vector3D object.
+    """
+    return Vector3D(vector.X, vector.Y, vector.Z)
+
+
+def remove_dup_vertices(vertices, tolerance):
+    """Remove vertices from an array of Point3Ds that are equal within the tolerance.
+
+    Args:
+        vertices: A list of Ladybug Point3D objects.
+
+    Returns:
+         A list of Ladybug Point3D objects with duplicate points removed.
+
+    """
+    return [pt for i, pt in enumerate(vertices)
+            if not pt.is_equivalent(vertices[i - 1], tolerance)]
+
+
+def mesh_to_face3d(mesh):
+    """Get a list of Ladybug Face3D objects from a rhino3dm Mesh.
+
+    Args:
+        mesh: A rhino3dm mesh geometry.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+
+    faces = []
+    pts = [to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices))]
+
+    for j in range(len(mesh.Faces)):
+        face = mesh.Faces[j]
+        if len(face) == 4:
+            all_verts = (pts[face[0]], pts[face[1]],
+                         pts[face[2]], pts[face[3]])
+        else:
+            all_verts = (pts[face[0]], pts[face[1]], pts[face[2]])
+
+        faces.append(Face3D(all_verts))
+
+    return faces
+
+
+def extract_mesh_faces_colors(mesh, color_by_face=False):
+    """Extract face indices and colors from a Rhino mesh.
+
+    Args:
+        mesh: Rhino3dm mesh
+        color_by_face: Bool. True will derive colors from mesh faces
+
+    Returns:
+        A tuple of mesh faces and mesh colors
+    """
+
+    colors = None
+    lb_faces = []
+    for i in range(len(mesh.Faces)):
+        face = mesh.Faces[i]
+        if len(face) == 4:
+            lb_faces.append((face[0], face[1], face[2], face[3]))
+        else:
+            lb_faces.append((face[0], face[1], face[2]))
+    if len(mesh.VertexColors) != 0:
+        colors = []
+        if color_by_face is True:
+            for j in range(len(mesh.Faces)):
+                face = mesh.Faces[j]
+                col = mesh.VertexColors[face[0]]
+                colors.append(lbc.Color(col.R, col.G, col.B))
+        else:
+            for k in range(len(mesh.VertexColors)):
+                col = mesh.VertexColors[k]
+                colors.append(lbc.Color(col.R, col.G, col.B))
+    return lb_faces, colors
+
+
+def mesh_to_mesh3d(mesh, color_by_face=True):
+    """Get a Ladybug Mesh3D object from a Rhino3dm Mesh.
+
+    Args:
+        mesh: Rhino3dm mesh
+        color_by_face: Bool. True will derive colors from mesh faces
+
+    Returns:
+        A Ladybug Mesh3D object
+    """
+    lb_verts = tuple(to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices)))
+    lb_faces, colors = extract_mesh_faces_colors(mesh, color_by_face)
+    return Mesh3D(lb_verts, lb_faces, colors)
+
+
+def brep_to_face3d(brep):
```

Comment:
The name for this method doesn't really seem appropriate. Technically, you're converting it to a list of triangulated Face3Ds or meshed Face3Ds.  What about `brep_to_meshed_face3d`?  Or, if you would rather not change the name, at least update the docstring to make it clear that this method uses the meshing pathway.

Reply:
Complied. I renamed the function.